### PR TITLE
Moved lease related operations to DataLakeLeaseClient/ShareLeaseClient.

### PR DESCRIPTION
**Pull request by katmsft**
Resolves https://github.com/Azure/azure-sdk-for-cpp/issues/1426
Resolves https://github.com/Azure/azure-sdk-for-cpp/issues/1425

# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

File: sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "azure/storage/files/datalake/datalake_lease_client.hpp"
+
+namespace Azure { namespace Storage { namespace Files { namespace DataLake {
+
+  const std::chrono::seconds DataLakeLeaseClient::InfiniteLeaseDuration{
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp:8]: I don't know if this works. Google `static initialization order fiasco`

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    {
+    }
+
+    /**
+     * @brief Initializes a new instance of the ShareLeaseClient.
+     *
+     * @param shareClient A ShareClient representing the share being leased.
+     * @param leaseId A lease ID. This is not required for break operation.
+     */
+    explicit ShareLeaseClient(ShareClient shareClient, std::string leaseId)
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:35]: Does share support lease operations? https://docs.microsoft.com/en-us/rest/api/storageservices/operations-on-shares--file-service-

File: sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+      Blobs::BlobLeaseClient::InfiniteLeaseDuration};
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp:9]:
```suggestion
  const std::chrono::seconds DataLakeLeaseClient::InfiniteLeaseDuration{-1};
```

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:35]: Yes, take a look at what .NET does: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/storage/Azure.Storage.Files.Shares/src/ShareLeaseClient.cs#L34

File: sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp (hunk 1 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-datalake/src/datalake_lease_client.cpp:8]: You are actually right, didn't realize that.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:35]: If you cover it with unit test cases and it actually works, then i'm fine even if it's not documented.

**Review by Jinming-Hu — APPROVED** [review]

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp (hunk 2 above)
Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp]: Seems the share lease still not ready on service end. Moved it to private just as .NET.

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+        const AcquireShareLeaseOptions& options = AcquireShareLeaseOptions());
+
+    /**
+     * @brief Renews the file or share's previously-acquired lease.
+     *
+     * @param leaseId ID of the previously-acquired lease.
+     * @param options Optional parameters to execute this function.
+     * @return A RenewShareLeaseResult describing the lease.
+     */
+    Azure::Core::Response<Models::RenewShareLeaseResult> Renew(
```
Comment (Jinming-Hu) [inline sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp:69]: move to private or just delete it

File: sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp (hunk 2 above)
Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp]: Do you know in which STG will this feature be available?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp]: If it won't be available in the foreseeable future, I tend to delete them, including constructor, renew and test cases.

Comment (katmsft) [reply sdk/storage/azure-storage-files-shares/inc/azure/storage/files/shares/share_lease_client.hpp]: It was said to be moved to STG75, I assume it is not turned on yet.

**Pull request merged**